iaso/api/payments/views.py

```diff
@@ -0,0 +1,156 @@
+from django.db.models import Count
+import django_filters
+from drf_yasg import openapi
+from rest_framework import filters, permissions
+from rest_framework.exceptions import NotFound
+from hat.menupermissions import models as permission
+from iaso.api.common import (
+    HasPermission,
+    ModelViewSet,
+)
+from iaso.models import Payment, OrgUnitChangeRequest, PotentialPayment, OrgUnit
+import iaso.api.payments.filters as potential_payment_filters
+from .serializers import PotentialPaymentSerializer
+from drf_yasg.utils import swagger_auto_schema
+
+
+class PotentialPaymentsViewSet(ModelViewSet):
+    """
+    # `Potential payment` API
+
+    This API allows to list potential payments linked to multiple `OrgUnitChangeRequest` by the same user to be updated and queried.
+
+    The Django model that stores "Potential payment" is `PotentialPayment`.
+
+    Whenever the list endpoint is invoked, it evaluates whether a new change request can be incorporated into the potential payment, or if there's a need to generate a new potential payment.
+
+    ## Permissions
+
+    - User must be authenticated
+    - User needs `iaso_payments` permission
+
+    ## Possible responses
+
+    ### 200 - OK
+
+    ### 400 - Bad request
+
+    - `page` or `limit` cannot be parsed to a correct integer value
+
+    ### 401 - Unauthorized
+
+    - No authentication token or an invalid one was provided
+
+    ### 403 - Forbidden
+
+    - User doesn't have the proper permission to access this resource.
+
+
+    ### 404 - Not found
+
+    - `users`, `user_roles`, `parent_id` not found
+
+    """
+
+    permission_classes = [permissions.IsAuthenticated, HasPermission(permission.PAYMENTS)]
+    filter_backends = [
+        filters.OrderingFilter,
+        django_filters.rest_framework.DjangoFilterBackend,
+        potential_payment_filters.UsersFilterBackend,
+        potential_payment_filters.UserRolesFilterBackend,
+        potential_payment_filters.FormsFilterBackend,
+        potential_payment_filters.ParentFilterBackend,
+        potential_payment_filters.StartEndDateFilterBackend,
+    ]
+    ordering_fields = [
+        "user__username",
+        "user__last_name",
+        "user__first_name",
+        "created_at",
+        "updated_at",
+        "status",
+        "created_by__username",
+        "updated_by__username",
+        "change_requests",
+    ]
+    serializer_class = PotentialPaymentSerializer
+
+    results_key = "results"
+    http_method_names = ["get", "head", "options", "trace"]
+
+    def get_queryset(self):
+        queryset = PotentialPayment.objects.all()
```

Comment:
You are not obliged to do `queryset = queryset` .. each time, you can chain. and you don't need to start with all().